engine/src/main/java/org/terasology/world/propagation/StandardBatchPropagator.java

```diff
@@ -272,9 +261,9 @@ private void increase(Vector3i position, byte value) {
      * @param position The position to set at
      * @param oldValue The original value at the position
      */
-    private void reduce(Vector3i position, byte oldValue) {
+    private void reduce(Vector3ic position, byte oldValue) {
         if (oldValue > 0) {
-            reduceQueues[rules.getMaxValue() - oldValue].add(position);
+            reduceQueues[rules.getMaxValue() - oldValue].add(new Vector3i(position));
```

Comment:
Make a copy when storing the vector in local persistence.

Reply:
that should avoid bugs if the value is modified outside :+1: